Realm/RLMSyncConfiguration.mm

```diff
@@ -270,32 +285,43 @@ - (BOOL)enableFlexibleSync {
     return make_sync_error(errorClass, @(error.message.c_str()), error.error_code.value(), custom);
 }
 
-static void setDefaults(SyncConfig& config, RLMUser *user) {
-    config.client_resync_mode = ClientResyncMode::Manual;
-    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
-
+- (void)setErrorHandler:(RLMUser *)user {
     RLMSyncManager *manager = [user.app syncManager];
     __weak RLMSyncManager *weakManager = manager;
-    config.error_handler = [weakManager](std::shared_ptr<SyncSession> errored_session, SyncError error) {
+    _config->error_handler = [weakManager, self](std::shared_ptr<SyncSession> errored_session, SyncError error) {
         RLMSyncErrorReportingBlock errorHandler;
         @autoreleasepool {
             errorHandler = weakManager.errorHandler;
         }
-        if (!errorHandler) {
```

Comment:
why remove this, I would probably first check if both errorHandler and _manualClientResetHandler are not null, in that case is not even worth it to check for the error

Reply:
https://github.com/realm/realm-swift/pull/7938/files#r965220073